### PR TITLE
update golang version of project and testimages to 1.19

### PR DESCRIPTION
**Pull request by RiSKeD**
To include native golang fuzzing tests in u-root, a version bump to golang version 1.18 is necessary.
The testimage-amd64 Dockerfile uses the latest commit of [multiboot-test-kernel](https://github.com/u-root/multiboot-test-kernel/commit/1c7e4f4722077dcab308cd1df9818eab011e58c4) with updated build flags.

Signed-off-by: Fabian Wienand <fabian.wienand@9elements.com>

### CODE REVIEW

**Comment by RiSKeD** [comment]
Will remain in draft until I am able to push the updated docker images to docker hub.

**Comment by hugelgupf** [comment]
@rminnich one last release on 1.17 before the bump?

**Comment by rminnich** [comment]
so ... how do you folks feel about 1.20? It has the new error trees instead of error chains stuff. Or at least the PR went in ... maybe we need 1.21?

multierror is SO MUCH NICER with error trees.

**Comment by MDr164** [comment]
> so ... how do you folks feel about 1.20?

I mean go 1.20 is scheduled for February 2023, is your plan to keep this PR on halt until then to skip directly to 1.20? I think 1.18 is a reasonable bump as 1.19 is the current latest release and quite a few potentially necessary things got added in 1.18. I mean improved error handling is nice, but I don't think it's holding out for a version bump that long if 1.18 brings things we could need.

Or am I misinterpreting the comment?

**Comment by RiSKeD** [comment]
@rminnich I built the docker images and ran the tests for go 1.19 locally as well. It really depends on if we want to move up to the latest version to be up-to-date or stick to 1.18 to enable more functionality. Either way, in my opinion bumping up now to at least go 1.18 comes with new opportunities like the native golang fuzzing support.

**Comment by RiSKeD** [comment]
Fixed up a small issue regarding the working directory because of the newer [cimg/go image](https://discuss.circleci.com/t/legacy-convenience-image-deprecation/41034#go-8).

**Comment by RiSKeD** [comment]
@rminnich There is a test TestNoActionDoubleWait in the [syscallfilter pkg](https://github.com/RiSKeD/u-root/blob/golang-version-bump-to-1.18/pkg/syscallfilter/syscallfilter_linux_test.go), which never returns due to a change in the handling of [Wait() in the os/exec pkg](https://github.com/golang/go/commit/4e79f06dac712d35d67d72777dae6df6d8c97888) for go1.19. The expected error in the test is never returned because of this and just idles. Could you take a look at it?

**Comment by rminnich** [comment]
see https://github.com/u-root/u-root/pull/2519

File: cmds/boot/pxeboot/pxeboot.go [hunk 1]
```diff
@@ -11,12 +11,12 @@
 // ServerName options (which may be embedded in the original BOOTP message, or
 // as option codes) to find something to boot.
 //
-// This BootFileName may point to
+// # This BootFileName may point to
```
Comment (rminnich) [inline cmds/boot/pxeboot/pxeboot.go:14]: Is the # here needed?

File: cmds/boot/pxeboot/pxeboot.go [hunk 2]
```diff
@@ -11,12 +11,12 @@
...
 //
 // - an iPXE script beginning with #!ipxe
 //
-// - a pxelinux.0, in which case we will ignore the pxelinux and try to parse
-//   pxelinux.cfg/<files>
+//   - a pxelinux.0, in which case we will ignore the pxelinux and try to parse
```
Comment (rminnich) [inline cmds/boot/pxeboot/pxeboot.go:18]: why the extra space? Did something automangle it?

**Review by rminnich — DISMISSED** [review]
this is a hard review given the insane amount of seemingly unnecessary reformatting ... The only thing that puzzled me is the # in the comment in that one go file. I guess the reformatting comes with the territoty.

File: cmds/boot/pxeboot/pxeboot.go (hunk 1 above)
Comment (RiSKeD) [reply cmds/boot/pxeboot/pxeboot.go:14]: See [Go1.19 Release Notes - Doc Comments](https://go.dev/doc/go1.19). This line is interpreted as a heading under the new syntax. I will push a fix.

**Review by MDr164 — APPROVED** [review]
Not sure if the go doc change in 1.19 caused you to reformat the package docs but I went through all of them and the changes all look sane, The content of the docs is unmodified and they are all uniform now to the go package doc spec. As the CI runs through and the changes to bump us to 1.19 are minimal I'll approve this PR. I also recently marked a new tag (v0.10.0) in case you want to jump back to 1.17

**Pull request merged**